lib/bolt/transport/api.rb

```diff
@@ -7,21 +7,26 @@
 module Bolt
   module Transport
     class Api < Orch
-      def initialize(*args)
-        super
+      attr_reader :plan_job
+
+      def initialize(plan_job, *args)
+        @plan_job = plan_job
+        @connection = nil
+        super(*args)
       end
 
       def get_connection(conn_opts)
-        key = Bolt::Transport::Api::Connection.get_key(conn_opts)
-        unless (conn = @connections[key])
-          @connections[key] = Bolt::Transport::Api::Connection.new(conn_opts, logger)
-          conn = @connections[key]
-        end
-        conn
+        @connection ||= Bolt::Transport::Api::Connection.new(conn_opts, logger, @plan_job)
       end
 
       def batches(targets)
-        targets.group_by { |target| Bolt::Transport::Api::Connection.get_key(target.options) }.values
+        targets
+      end
+
+      def finish_plan(result)
+        @connection.finish_plan(result)
```

Comment:
This fails if the plan doesn't run any tasks, since `run_task_job` is the spot where `get_connection` is originally called. Otherwise, `@connection` is nil.